### PR TITLE
feat: Add new powerMonitor synchronous API 

### PR DESCRIPTION
**Pull request by nitsakh**
#### Description of Change
`powerMonitor.querySystemIdleState` and `powerMonitor.querySystemIdleTime` had async backing APIs in chromium (https://chromium-review.googlesource.com/c/chromium/src/+/1379183). However, that has changed in ch73. So, this PR deprecates the old async APIs and adds new sync APIs.

#### Checklist

- [x] PR description included and stakeholders cc'd
- [x] `npm test` passes
- [x] tests are [changed or added](https://github.com/electron/electron/blob/master/docs/development/testing.md)
- [x] relevant documentation is changed or added
- [x] PR title follows semantic [commit guidelines](https://github.com/electron/electron/blob/master/docs/development/pull-requests.md#commit-message-guidelines)
- [x] [PR release notes](https://github.com/electron/clerk/blob/master/README.md) describe the change in a way relevant to app developers, and are [capitalized, punctuated, and past tense](https://github.com/electron/clerk/blob/master/README.md#examples).

#### Release Notes

Notes: Replaced `powerMonitor` `querySystemIdleState` and `querySystemIdleTime` with synchronous APIs.

### CODE REVIEW

File: docs/api/breaking-changes.md [hunk 1]
```diff
@@ -71,6 +71,24 @@ Child windows opened with the `nativeWindowOpen` option will always have Node.js
 Renderer process APIs `webFrame.setRegisterURLSchemeAsPrivileged` and `webFrame.registerURLSchemeAsBypassingCSP` as well as browser process API `protocol.registerStandardSchemes` have been removed.
 A new API, `protocol.registerSchemesAsPrivileged` has been added and should be used for registering custom schemes with the required privileges. Custom schemes are required to be registered before app ready.
 
+## `powerMonitor.querySystemIdleState`
+
+```js
+// Deprecated
+powerMonitor.querySystemIdleState(threshold, callback)
+// Replace with synchronous API
+idleState = getSystemIdleState(threshold)
```
Comment (codebytere) [inline docs/api/breaking-changes.md]: for clarity, probably best to have:
```js
// Deprecated
powerMonitor.querySystemIdleState(threshold, callback)
// Replace with synchronous API
const idleState = powerMonitor.getSystemIdleState(threshold)
```

File: docs/api/breaking-changes.md [hunk 2]
```diff
@@ -71,6 +71,24 @@ Child windows opened with the `nativeWindowOpen` option will always have Node.js
...
+```
+
+## `powerMonitor.querySystemIdleTime`
+
+```js
+// Deprecated
+powerMonitor.querySystemIdleTime(callback)
```
Comment (codebytere) [inline docs/api/breaking-changes.md]: ditto here as the above

File: spec/api-power-monitor-spec.js [hunk 3]
```diff
@@ -163,5 +165,37 @@ describe('powerMonitor', () => {
...
+        const idleState = powerMonitor.getSystemIdleState(1)
+        expect(idleState).to.be.a('string')
+        const validIdleStates = [ 'active', 'idle', 'locked', 'unknown' ]
+        expect(validIdleStates).to.include(idleState)
+      })
+
+      it('does not accept non positive integer threshold', () => {
+        expect(() => {
+          powerMonitor.getSystemIdleState(-1)
+        }).to.throw()
```
Comment (codebytere) [inline spec/api-power-monitor-spec.js]: do you know what it throws with? we may want to be more specific if possible!

**Comment by deepak1556** [comment]
Lets avoid alternate api for this case. We can run the deprecation in 73 branch, its easier to turn sync implementation to async and reliable https://github.com/electron/electron/pull/16494/commits/4a58cc3aabd729e2a5f703e70758c5c262d1fb95.

**Pull request closed without merging**